src/Cli/dotnet/commands/dotnet-test/VSTestArgumentConverter.cs

```diff
@@ -148,14 +177,39 @@ public List<string> Convert(string[] args, out List<string> ignoredArgs)
                     {
                         UpdateVerbosity(arg, newArgList);
                     }
-                    else if (IgnoredArguments.Contains(activeArgument))
+                    // This case should be processed before 's_ignoredArguments.Contains'.
+                    else if (s_switchArguments.Contains(activeArgument))
+                    {
+                        if (s_ignoredArguments.Contains(activeArgument))
+                        {
+                            ignoredArgs.Add(activeArgument);
+                        }
+                        else
+                        {
+                            newArgList.Add(activeArgument);
+                        }
+
+                        // The only real case where we would end-up here is when the item under test (.dll or .exe)
+                        // is passed as last argument so we could potentially check the extension and for invalid ones,
+                        // either add it to the ignored args or have a specific failure.
+                        newArgList.Add(arg);
+                    }
+                    else if (s_ignoredArguments.Contains(activeArgument))
                     {
                         ignoredArgs.Add(activeArgument);
                         ignoredArgs.Add(arg);
                     }
-                    else if (blame.IsBlameArg(activeArgument, arg))
+                    else if (BlameArgs.IsBlameArg(activeArgument))
```

Comment:
Maybe explain here (if that is the correct place) or in the called method what is the problem with --blame and --blame CollectDump. I know what the problem is, but I still can't see what is going on here.

Reply:
Let me know if that's clearer now.